### PR TITLE
8274280: regenarate libclang binding in jextract module

### PR DESCRIPTION
**Pull request by sundararajana**
Moved CLayouts out of the generated code package. Except for libName & copyright header, no changes to the jextract generated code now.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

### Issue
 * [JDK-8274280](https://bugs.openjdk.java.net/browse/JDK-8274280): regenarate libclang binding in jextract module

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.java.net/census#mcimadamore) (@mcimadamore - Committer) ⚠️ Review applies to 13e06b995571e6487b89dff2edd39b9e3ac8639d

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/panama-foreign pull/580/head:pull/580` \
`$ git checkout pull/580`

Update a local copy of the PR: \
`$ git checkout pull/580` \
`$ git pull https://git.openjdk.java.net/panama-foreign pull/580/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 580`

View PR using the GUI difftool: \
`$ git pr show -t 580`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/panama-foreign/pull/580.diff">https://git.openjdk.java.net/panama-foreign/pull/580.diff</a>

</details>

### CODE REVIEW

File: src/jdk.incubator.jextract/share/classes/jdk/internal/clang/libclang/CXToken.java [hunk 1]
```diff
@@ -31,15 +31,12 @@
...
-
+import static jdk.incubator.foreign.ValueLayout.*;
 public class CXToken {
 
     static final MemoryLayout $struct$LAYOUT = MemoryLayout.structLayout(
-        MemoryLayout.sequenceLayout(4, C_INT).withName("int_data"),
-        C_POINTER.withName("ptr_data")
+        MemoryLayout.sequenceLayout(4, JAVA_INT).withName("int_data"),
+        ADDRESS.withName("ptr_data")
     );
```
Comment (JornVernee) [inline src/jdk.incubator.jextract/share/classes/jdk/internal/clang/libclang/CXToken.java:40]: Why these changes? Shouldn't this still be using `C_INT` and `C_POINTER` ?

Comment (sundararajana) [reply src/jdk.incubator.jextract/share/classes/jdk/internal/clang/libclang/CXToken.java:40]: That's generated code. In this patch I am not touching jextract generated code except for copyright header + libName

File: src/jdk.incubator.jextract/share/classes/jdk/internal/clang/CLayouts.java [hunk 2]
```diff
@@ -23,11 +23,11 @@
  *  questions.
  */
 
-package jdk.internal.clang.libclang;
+package jdk.internal.clang;
```
Comment (mcimadamore) [inline src/jdk.incubator.jextract/share/classes/jdk/internal/clang/CLayouts.java]: This class should be removed. Other classes in the libclang package should use the C layouts that are generated in the main class header. Outside of the package, we should just use regular ValueLayouts (e.g. `ADDRESS`).

**Review by mcimadamore — APPROVED** [review]
Looks good!
